### PR TITLE
added the possibility of passing around an object with arbitrary data

### PR DESCRIPTION
**Pull request by dbarrosop**
When instantiating a parser/translator you can pass an arbitrary dict via the parameter `extra`:

https://github.com/networktocode/yangify/pull/12/files#diff-a6e2a0866ba4b2ee96b797b1ed04e13bR178

That dictionary will be available throughout the entire lifecycle of the object via `self.yy.extra`:

https://github.com/networktocode/yangify/pull/12/files#diff-a6e2a0866ba4b2ee96b797b1ed04e13bR73
https://github.com/networktocode/yangify/pull/12/files#diff-a6e2a0866ba4b2ee96b797b1ed04e13bR78

### CODE REVIEW

File: yangify/translator/__init__.py [hunk 1]
```diff
@@ -490,7 +504,16 @@ def __init__(
             o = None
 
         super().__init__(
-            None, None, instance.InstanceRoute(), dm, dm.schema, {}, n, o, replace
+            None,
```
Comment (jathanism) [inline yangify/translator/__init__.py]: Oof this is not the greatest readability. Would you mind using keyword arguments for these to improve this? (lines 507-516)

File: tests/unit/test_parser.py [hunk 2]
```diff
@@ -64,6 +64,36 @@ class RootTestParser(parser.RootParser):
     start = StartParser
 
 
+class RootTestParserWithExtra(parser.RootParser):
```
Comment (jathanism) [inline tests/unit/test_parser.py]: It's not really clear what's going on here with all of these nested inner classes. Can you please provide a brief docstring or comments (or both?) on how this works?

File: tests/unit/test_parser.py [hunk 3]
```diff
@@ -142,3 +172,55 @@ def test_parse_state(self) -> None:
                 }
             }
         }
+
+    def test_parse_config_with_extra(self) -> None:
+        parser = RootTestParserWithExtra(
+            dm, test_data, config=True, state=False, extra={"os_version": "test"}
+        )
+        parsed_obj = parser.process()
+        assert parsed_obj.raw_value() == {
```
Comment (jathanism) [inline tests/unit/test_parser.py]: It looks like this payload (lines 181-200) and the payload used in `.test_parse_state_with_extra()` (lines 207-226) are identical except for the `config` and `state` words. Would a variable fixture be better here to reduce duplication and improve readability?

File: tests/unit/test_translator.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+from yangson.datamodel import DataModel
+
+
+BASE = pathlib.Path(__file__).parent
+dm = DataModel.from_file(
+    f"{BASE}/yang/simple/yang-library-data.json", [f"{BASE}/yang/simple/"]
+)
+
+
+test_data = {
```
Comment (jathanism) [inline tests/unit/test_translator.py]: This looks like the same fixture from `test_parser.py` (lines 181-200). Can we define these fixtures centrally for use across various test modules?

File: tests/unit/test_parser.py (hunk 3 above)
Comment (dbarrosop) [reply tests/unit/test_parser.py]: Not a fixture but a global variable like `test_data` might help. Will do.

File: tests/unit/test_translator.py (hunk 4 above)
Comment (dbarrosop) [reply tests/unit/test_translator.py]: Don't like fixtures for data, makes things harder to read as now you have to navigate between files to understand the test. Also, tests are completely independent, they have the same data now because it was easier to write this way but if we find edge cases we want to test they may evolve independently.

File: yangify/translator/__init__.py (hunk 1 above)
Comment (dbarrosop) [reply yangify/translator/__init__.py]: fixed

File: tests/unit/test_parser.py (hunk 2 above)
Comment (dbarrosop) [reply tests/unit/test_parser.py]: fixed

File: tests/unit/test_parser.py (hunk 3 above)
Comment (dbarrosop) [reply tests/unit/test_parser.py]: fixed

**Comment by dbarrosop** [comment]
If you don't have any more comments feel free to merge yourself :)

File: tests/unit/test_translator.py (hunk 4 above)
Comment (jathanism) [reply tests/unit/test_translator.py]: I accept your answer and will move on with my life.
